Angular/src/app/app.component.html

```diff
@@ -0,0 +1,35 @@
+<div>
+  <dx-scheduler
+    (onAppointmentAdding)='handleAppointmentActions($event, $event.appointmentData)'
+    (onAppointmentUpdating)='handleAppointmentActions($event, $event.newData)'
```

Comment:
See the comment about handling updating an existing recurrent appointment case in the app.component.ts:

```suggestion
    (onAppointmentAdding)='handleAppointmentAdd($event)'
    (onAppointmentUpdating)='handleAppointmentUpdate($event)'
```